app/src/main/java/net/bible/service/db/readingplan/ReadingPlanDatabaseDefinition.kt

```diff
@@ -0,0 +1,499 @@
+import net.bible.android.SharedConstants
+import net.bible.service.common.CommonUtils
+import net.bible.service.readingplan.PassageReader
+
+import org.apache.commons.lang3.StringUtils
+import org.apache.commons.lang3.StringUtils.trim
+import org.crosswire.common.util.IOUtil
+import org.crosswire.jsword.passage.PassageKeyFactory
+import org.crosswire.jsword.versification.Versification
+import org.crosswire.jsword.versification.system.SystemKJV
+import org.crosswire.jsword.versification.system.SystemNRSVA
+import org.crosswire.jsword.versification.system.Versifications
+import java.io.BufferedReader
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FileInputStream
+import java.io.InputStream
+import java.io.IOException
+import java.util.Properties
+import kotlin.collections.ArrayList
+
+/** @author Timmy Braun [tim.bze at gmail dot com] (2/12/2019)
+ */
+object ReadingPlanDatabaseDefinition {
+
+    object ReadingPlanMeta : BaseColumns {
+        // If COLUMN_CURRENT_DAY has CONSTANT_CURRENT_DAY_BY_DATE value,
+        // then the reading plan will by default get today's date
+        const val CONSTANT_CURRENT_DAY_BY_DATE = -1
+
+        const val TABLE_NAME = "readingplan_meta"
+        const val COLUMN_ID = BaseColumns._ID
+        const val COLUMN_PLAN_FILE_NAME = "plan_file_name"
+        const val COLUMN_PLAN_NAME = "plan_name"
+        const val COLUMN_DESCRIPTION = "description"
+        const val COLUMN_DATE_START = "date_start"
+        const val COLUMN_DAYS_IN_PLAN = "days_in_plan"
+        const val COLUMN_CURRENT_DAY = "current_day"
+        const val COLUMN_VERSIFICATION_NAME = "versification"
+    }
+
+    object ReadingPlanDays : BaseColumns {
+        const val TABLE_NAME = "readingplan_days"
+        const val COLUMN_ID = BaseColumns._ID
+        const val COLUMN_READING_PLAN_META_ID = "readingplan_meta_id"
+        const val COLUMN_DAY_NUMBER = "day_number"
+        const val COLUMN_READING_DATE = "reading_date"
+        const val COLUMN_DAY_CHAPTERS = "day_chapters"
+        const val COLUMN_READ_STATUS = "read_status"
+    }
+
+    class Operations {
+        companion object {
+            private const val TAG = "ReadingPlanDatabaseOps"
+            private const val SQL_CREATE_ENTRIES_META =
+                """CREATE TABLE ${ReadingPlanMeta.TABLE_NAME} (
+                ${ReadingPlanMeta.COLUMN_ID} INTEGER PRIMARY KEY AUTOINCREMENT,
+                ${ReadingPlanMeta.COLUMN_PLAN_FILE_NAME} TEXT NOT NULL,
+                ${ReadingPlanMeta.COLUMN_PLAN_NAME} TEXT NOT NULL,
+                ${ReadingPlanMeta.COLUMN_DESCRIPTION} TEXT NOT NULL,
+                ${ReadingPlanMeta.COLUMN_DATE_START} INTEGER,
+                ${ReadingPlanMeta.COLUMN_DAYS_IN_PLAN} INTEGER,
+                ${ReadingPlanMeta.COLUMN_CURRENT_DAY} INTEGER NOT NULL DEFAULT 0,
+                ${ReadingPlanMeta.COLUMN_VERSIFICATION_NAME} TEXT NOT NULL
+                ); """
+
```

Comment:
I was experimenting with kotlin a little and found out that you could write this a bit neater like this:

```kt
            private val SQL_CREATE_ENTRIES_META = ReadingPlanMeta.run {
                """CREATE TABLE ${ReadingPlanMeta.TABLE_NAME} (
                ${COLUMN_ID} INTEGER PRIMARY KEY AUTOINCREMENT,
                ${COLUMN_PLAN_FILE_NAME} TEXT NOT NULL,
                ${COLUMN_PLAN_NAME} TEXT NOT NULL,
                ${COLUMN_DESCRIPTION} TEXT NOT NULL,
                ${COLUMN_DATE_START} INTEGER,
                ${COLUMN_DAYS_IN_PLAN} INTEGER,
                ${COLUMN_CURRENT_DAY} INTEGER NOT NULL DEFAULT 0,
                ${COLUMN_VERSIFICATION_NAME} TEXT NOT NULL
                ); """
            }
```

Reply:
If you like this, change all others similarly as well.

Reply:
Nice. Done